test/jaxp/javax/xml/jaxp/unittest/common/jdkcatalog/JDKCatalogTest.java

```diff
@@ -106,6 +131,19 @@ public Object[][] getXSD() throws Exception {
         };
     }
 
+    /**
+     * Verifies that the JDK built-in Catalog supports both the Public and System
+     * identifiers for DTDs.
+     * @param publicId the public Id
+     * @param systemId the system Id
+     */
+    @Test(dataProvider = "DTDsInJDKCatalog")
+    public void testDTDsInJDKCatalog(String publicId, String systemId) {
+        String matchingPubId = JDKCATALOG.matchPublic(publicId);
+        String matchingSysId = JDKCATALOG.matchSystem(systemId);
```

Comment:
Isn't there a public API to get these values. It seems a bit of a hack to have to break module encapsulation.

Reply:
Updated. Thanks!